tests/std/tests/P0896R4_ranges_alg_move_backward/test.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+struct int_wrapper {
+    int val                 = 10;
+    constexpr int_wrapper() = default;
+    constexpr int_wrapper(int x) : val{x} {}
+    constexpr int_wrapper(int_wrapper&& that) : val{exchange(that.val, -1)} {}
+    constexpr int_wrapper& operator=(int_wrapper&& that) {
+        val = exchange(that.val, -1);
+        return *this;
+    }
+    auto operator<=>(const int_wrapper&) const = default;
+};
+
+// Validate that move_backward_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::move_backward_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::move_backward(borrowed<false>{}, static_cast<int*>(nullptr))),
+    ranges::move_backward_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::move_backward(borrowed<true>{}, static_cast<int*>(nullptr))),
+    ranges::move_backward_result<int*, int*>>);
+
+static constexpr int expected_output[]      = {13, 42, 1729};
+static constexpr int expected_input[]       = {-1, -1, -1};
+static constexpr int expected_overlapping[] = {-1, 0, 1, 2};
+
+struct instantiator {
+    template <ranges::bidirectional_range R1, ranges::bidirectional_range R2>
+    static constexpr void call() {
+#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, VSO-938163
+#pragma warning(suppress : 4127) // conditional expression is constant
+        if (!ranges::contiguous_range<R1> || !ranges::contiguous_range<R1> || !is_constant_evaluated())

```

Comment:
```suggestion
        if (!ranges::contiguous_range<R1> || !ranges::contiguous_range<R2> || !is_constant_evaluated())
```

Reply:
Ha! I didn't think this through. This isn't a "swapping" algorithm - which would pass source and destination iterators to `iter_swap` - it's a "move" algorithm that passes source iterators to `iter_move`. It will trigger the bug only if `R1` has iterators-that-unwrap-to-pointers.